### ISSUE TITLE
JRE not detected when found JDKs, but not matching criteria

### ISSUE THREAD

**Issue by tomaszguzialek** [tofi86/universalJavaApplicationStub#15]
The command /usr/libexec/java_home reports only JDKs and not JREs. It is a second priority choice after JAVA_HOME. Therefore, if we have JDK 1.6 reported by /usr/libexec/java_home and we set in our .plist file that we accept only 1.7+, the public JRE fallback will not be found and used (although we have JRE 1.8 available there).

**Comment by tofi86** [comment]
It seems to be a policy by oracle as discussed in another issue: if you need Java for anything else than the web browser plugin, you have to install the JDK.
That's for sure super insane, but that's what oracle says...

JRE's can't set neither the JAVA_HOME environment variable nor an entry in /usr/libexec/java_home

So what's your suggestion? To check the web browser plugins first? Then you'll never be able to specify JRE requirements...

It's all a mess what Oracle does with Java for Mac lately...

**Comment by tomaszguzialek** [comment]
Oh yeah, very much confusing when coming from Windows world to Mac... Of course,from the perspective of the script maintainer, there is no possibility to control whether JRE will be available in any of the paths / entries. I won't then suggest changing the priorities of the search locations.

However, currently when some JDK is detected (1.6) in /usr/libexec/java_home, but it does not match the version specified (1.7+), an error message is shown right away:
```
# check for specified JVMversion in "/usr/libexec/java_home" symlinks
elif [ ! -z ${JVMVersion} ] && [ -x /usr/libexec/java_home ] && /usr/libexec/java_home -F; then

	if /usr/libexec/java_home -F -v ${JVMVersion}; then
		JAVACMD="`/usr/libexec/java_home -F -v ${JVMVersion} 2> /dev/null`/bin/java"
	else
		# display error message with applescript
		osascript -e "tell application \"System Events\" to display dialog \"ERROR launching '${CFBundleName}'\n\nNo suitable Java version found on your system!\nThis program requires Java ${JVMVersion}\nMake sure you install the required Java version.\" with title \"${CFBundleName}\" buttons {\" OK \"} default button 1 with icon path to resource \"${CFBundleIconFile}\" in bundle (path to me)"
		# exit with error
		exit 3
	fi
```
Shouldn't we treat it like no JDK is found at all and check other fallback options (JRE plugin)?

**Comment by tofi86** [comment]
this is This is going to be fixed in the next version 1.0.0. Stable release isn't out yet, but code will be on master branch in a minute.
Please go ahead and test!

Now with this change, if in Info.plist a specific JVM version is required, the script first checks `/usr/libexec/java_home -F -v ${JVMVersion}` to find a suitable JDK, if that fails, it checks the Apple JRE in `/Library/Java/Home/`, if that fails it checks the Oracle JRE in `/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/` and only if all of these fail to return a suitable Java version, the applescript error message is shown.
